packages/elements/image-ui/src/ImageElement/ImageElement.tsx

```diff
@@ -1,32 +1,146 @@
   const focused = useFocused();
   const selected = useSelected();
+  const editor = useEditorRef();
+
+  const [width, setWidth] = useState(nodeWidth);
+
+  useEffect(() => {
+    setWidth(nodeWidth);
+  }, [nodeWidth]);
 
   const styles = getImageElementStyles({ ...props, focused, selected });
 
+  const setNodeWidth = useCallback(
+    (w: number) => {
+      const path = ReactEditor.findPath(editor, element);
+
+      if (w === nodeWidth) {
+        // Focus the node if not resized
+        Transforms.select(editor, path);
+      } else {
+        setNodes(editor, { width: w }, { at: path });
+      }
+    },
+    [editor, element, nodeWidth]
+  );
+
+  const onChangeCaption = useCallback(
+    (e: any[]) => {
+      const path = ReactEditor.findPath(editor as ReactEditor, element);
+      setNodes(editor, { caption: e }, { at: path });
+    },
+    [editor, element]
+  );
+
   return (
     <div
       {...attributes}
       css={styles.root.css}
       className={styles.root.className}
     >
       <div contentEditable={false}>
-        <img
-          data-testid="ImageElementImage"
-          css={styles.img?.css}
-          className={styles.img?.className}
-          src={url}
-          alt=""
-          {...nodeProps}
-        />
+        <figure
+          css={styles.figure?.css}
+          className={`group ${styles.figure?.className}`}
+        >
+          <Resizable
+            size={{ width, height: '100%' }}
+            maxWidth="100%"
+            lockAspectRatio
+            enable={{ right: true }}
+            handleComponent={{
+              right: (
+                <ImageHandle
+                  css={styles.handle?.css}
+                  className={styles.handle?.className}
+                />
+              ),
+            }}
+            handleStyles={{
+              right: {
+                right: 0,
+              },
+            }}
+            onResize={(e, direction, ref) => {
+              setWidth(ref.offsetWidth);
+            }}
+            onResizeStop={(e, direction, ref) => setNodeWidth(ref.offsetWidth)}
+            {...resizableProps}
+          >
+            <img
+              data-testid="ImageElementImage"
+              css={styles.img?.css}
+              className={styles.img?.className}
+              src={url}
+              alt={caption}
+              {...nodeProps}
+            />
+          </Resizable>
+          {!disableCaption && (caption.length || selected) && (
+            <figcaption style={{ width }}>
+              <div css={styles.captionInput?.css}>
+                <Plate
+                  id="caption-editor"
+                  plugins={plugins}
+                  initialValue={caption}
+                  value={caption}
+                  editableProps={{
+                    placeholder: captionPlaceholder,
+                    className: styles.captionInput?.className,
+                  }}
+                  onChange={onChangeCaption}
+                />
```

Comment:
I added the Plate editor for the caption. Since the node doesn't have any id how can we pass separate id to the caption editor? @zbeyens

Reply:
I've added nanoid to solve that

Reply:
Did you add in the plate?

Reply:
It seams the package is not used in the plate.